src/EventEdition/EventDialog.vala

```diff
@@ -214,9 +214,14 @@ public class EventDialog : Granite.Dialog {
         }
 
         private void remove_event () {
-            var calmodel = Calendar.EventStore.get_default ();
-            calmodel.remove_event (original_source, ecal, mod_type);
-            this.destroy ();
+            assert (original_source != null);
+            var delete_dialog = new Calendar.DeleteEventDialog (original_source, ecal, mod_type) {
+                transient_for = this
+            };
+            var response = delete_dialog.run_dialog ();
```

Comment:
Its a bit unusual for the dialog to destroy itself? Normally the process that creates it does that.